functions/common/function-test-support/src/main/java/org/springframework/cloud/fn/test/support/xmpp/XmppContainerSupport.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2014-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.test.support.xmpp;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.testcontainers.containers.DockerComposeContainer;
+
+import org.springframework.core.io.ClassPathResource;
+
+public abstract class XmppContainerSupport {
+
+	/**
+	 * Default XMPP Host.
+	 */
+	public static final String XMPP_HOST = "localhost";
+
+	/**
+	 * Default XMPP Port.
+	 */
+	public static final int XMPP_PORT = 5222;
+
+	/**
+	 * Sample John User setup by config.
+	 */
+	public static final String JOHN_USER = "john";
+
+	/**
+	 * Sample Jane User setup by config.
+	 */
+	public static final String JANE_USER = "jane";
+
+
+	/**
+	 * Password for sample users.
+	 */
+	public static final String USER_PW = "secret";
+
+	/**
+	 * Default Service Name.
+	 */
+	public static final String SERVICE_NAME = "localhost";
+
+	static final DockerComposeContainer XMPP_CONTAINER;
+
+	static {
+
+		try {
+
+			File dockerComposeFile = new ClassPathResource("/docker-compose/xmpp/docker-compose.yml").getFile();
```

Comment:
In relation to the fact of having to extract the content from the jar in order to reference it as a file...

I would really like to get rid of the maven plugin requirement. 

#### Option 1 - copy resource via in/out streams

To access the resource in all cases we have to get the input stream from the resource and then copy/transfer to the output stream. It looks something like this:

```java
File dockerComposeFile = File.createTempFile("my-docker-compose", ".yml");
ClassPathResource resource = new ClassPathResource("docker-compose/xmpp/docker-compose.yml");
try (InputStream inputStream = resource.getInputStream()) {
	try (FileOutputStream outputStream = new FileOutputStream(dockerComposeFile, false)) {
	  inputStream.transferTo(outputStream);
	  // OR FileCopyUtils.copy(inputStream, outputStream);
	}
}
System.out.println("***** " + dockerComposeFile.getAbsolutePath());
```
That is easy enough (works in both IDE and CLI), but the next issue is all of the conf that the docker-compose points to w/ its volume mount - it too has to have the same treatment. It can be done, but that is getting a bit complicated. 

#### Option 2 - create custom container / use built-in api 

There is a built-in API in `GenericContainer` that handles this for us `withClasspathResourceMapping`. We can create a custom XmppContainer, delete all use of docker-compose, solve the dynamic port issue, and the reosurce loading issue all in one shot. This got me really interested and I put together a WIP branch to illustrate. Please have a look and lmk what you think. https://github.com/spring-cloud/stream-applications/pull/382